ethcontract-derive/src/lib.rs

```diff
@@ -26,69 +27,166 @@ use syn::{
     Token, Visibility,
 };
 
-/// Proc macro to generate type-safe bindings to a contract. This macro accepts
-/// a path to a Truffle artifact JSON file. Note that this path is rooted in
-/// the crate's root `CARGO_MANIFEST_DIR`.
+/// Proc macro to generate type-safe bindings to a contract.
+///
+/// This macro accepts a path to an artifact JSON file. Note that this path
+/// is rooted in the crate's root `CARGO_MANIFEST_DIR`:
 ///
 /// ```ignore
-/// ethcontract::contract!("build/contracts/MyContract.json");
+/// contract!("build/contracts/WETH9.json");
 /// ```
 ///
 /// Alternatively, other sources may be used, for full details consult the
-/// `ethcontract-generate::source` documentation. Some basic examples:
+/// [`ethcontract_generate::source`] documentation. Some basic examples:
 ///
 /// ```ignore
 /// // HTTP(S) source
-/// ethcontract::contract!("https://my.domain.local/path/to/contract.json")
-/// // Etherscan.io
-/// ethcontract::contract!("etherscan:0x0001020304050607080910111213141516171819");
-/// ethcontract::contract!("https://etherscan.io/address/0x0001020304050607080910111213141516171819");
-/// // npmjs
-/// ethcontract::contract!("npm:@org/package@1.0.0/path/to/contract.json")
+/// contract!("https://my.domain.local/path/to/contract.json")
+///
+/// // etherscan.io
+/// contract!("etherscan:0xC02AAA39B223FE8D0A0E5C4F27EAD9083C756CC2");
+///
+/// // npm package
+/// contract!("npm:@openzeppelin/contracts@4.2.0/build/contracts/IERC20.json")
 /// ```
 ///
-/// Note that Etherscan rate-limits requests to their API, to avoid this an
+/// Note that etherscan rate-limits requests to their API, to avoid this an
```

Comment:
Shouldn't Etherscan be capitalized as it is a proper noun?

Reply:
I've done this to match capitalisation with other parts of code. I can change it separately.